pkg/p2p/peer/peer.go

```diff
@@ -159,7 +159,7 @@ func (w *Writer) Serve(writeQueueChan <-chan *bytes.Buffer, exitChan chan struct
 
 	// Any gossip topics are written into interrupt-driven ringBuffer
 	// Single-consumer pushes messages to the socket
-	g := &GossipConnector{w.gossip, w.Connection}
+	g := &GossipConnector{w.gossip, w.Connection, serviceFlag}
```

Comment:
I would use the `serviceFlag` to instantiate different `peer.Writer`s instead of modifying the `GossipConnector` logic depending on the flag